UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.UnitTests/Services/FssEventValidationAndMappingServiceTest.cs

```diff
@@ -0,0 +1,141 @@
+            FssEventData cloudEventData = JsonConvert.DeserializeObject<FssEventData>(data);
+
+            Assert.AreEqual(FssDataMappingValueConstant.Type, result.Type);
+            Assert.AreEqual(_fakeFssDataMappingConfiguration.Value.Source, result.Source);
+            Assert.AreEqual(batchDetailsUri, cloudEventData.Links.BatchDetails.Href);
+            Assert.AreEqual(batchDetailsUri + "/status", cloudEventData.Links.BatchStatus.Href);
+            Assert.AreEqual(batchDetailsUri + "/files/AVCS_S631-1_Update_Wk45_21_Only.zip", cloudEventData.Files.FirstOrDefault().Links.Get.Href);
+        }
+        #endregion
+
+        private static CustomCloudEvent GetCustomCloudEvent()
+        {
+            return new CustomCloudEvent()
+            {
+                Type = "uk.gov.UKHO.FileShareService.NewFilesPublished.v1",
+                Source = "https://files.admiralty.co.uk",
+                Id = "49c67cca-9cca-4655-a38e-583693af55ea",
+                Subject = "83d08093-7a67-4b3a-b431-92ba42feaea0",
+                DataContentType = "application/json",
+                Data = GetFssEventData(),
+                Time = "2021-11-09T14:52:28+00:00"
+            };
+        }
+
+        private static FssEventData GetFssEventData()
+        {
+            Link linkBatchDetails = new()
+            {
+                Href = @"https://files.admiralty.co.uk/batch/83d08093-7a67-4b3a-b431-92ba42feaea0"
+            };
+            Link linkBatchStatus = new()
+            {
+                Href = @"https://files.admiralty.co.uk/batch/83d08093-7a67-4b3a-b431-92ba42feaea0/status"
+            };
+
+            FileLinks fileLinks = new()
+            {
+                Get = new Link() { Href = @"https://files.admiralty.co.uk/batch/83d08093-7a67-4b3a-b431-92ba42feaea0/files/AVCS_S631-1_Update_Wk45_21_Only.zip" },
+            };
+
+            BatchLinks links = new()
+            {
+                BatchDetails = linkBatchDetails,
+                BatchStatus = linkBatchStatus
+            };
+
+            return new FssEventData()
+            {
+                Links = links,
+                BusinessUnit = "AVCSData",
+                Attributes = new List<Attribute> { },
+                BatchId = "83d08093-7a67-4b3a-b431-92ba42feaea0",
+                BatchPublishedDate = DateTime.UtcNow,
+                Files = new File[] {new() { MIMEType= "application/zip",
+                                                                    FileName= "AVCS_S631-1_Update_Wk45_21_Only.zip",
+                                                                    FileSize=99073923,
+                                                                    Hash="yNpJTWFKhD3iasV8B/ePKw==",
+                                                                    Attributes=new List<Attribute> {},
+                                                                    Links = fileLinks   }}
+            };
+        }
```

Comment:
These methods are identical in FssEventProcessorTest.cs and FssEventValidationAndMappingServiceTest.cs.  Can you move them into a base class or a static helper class please?

`GetFssEventData()` is also in WebhookControllerTest.cs, EnsEnterpriseEventServiceWebhookTest.cs and EventProcessorBaseTest.cs.

Reply:
Done changes added  static helper class.